src/CADL.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -583,20 +598,45 @@ function loadOperation(
             BodyMediaType: BodyMediaType.Json
         } as OperationResponse;
     }
+
+    function loadOperationLongRunning(program: Program, op: OperationDetails, resourceOperation?: ResourceOperation): OperationLongRunning | undefined {
+        if (!isLroOperation(program, op.operation)) return undefined;
+
+        let finalResponse = loadLongRunningFinalResponse(program, op, resourceOperation);
+        if (finalResponse === undefined) return undefined;
+
+        return {
+            FinalStateVia: OperationFinalStateVia.Location, // data plane only supports `location`
+            FinalResponse: finalResponse,
+        } as OperationLongRunning;
+    }
+
+    function loadLongRunningFinalResponse(program: Program, op: OperationDetails, resourceOperation?: ResourceOperation): OperationResponse | undefined {
+        for (const response of op.responses) {
+            if (response.statusCode == "200" || response.statusCode == "204")
```

Comment:
If there are  both `200` and `204` responses, what is expected behavior? throw error? or choose one?

Reply:
what is final response? how do we use it? Does every LRO have final response?

Reply:
> If there are both `200` and `204` responses, what is expected behavior? throw error? or choose one?

I think we prefer `200` which has some return content. `204` normally should not co-exist with `200`. If they both exist, `200` is better because it returns something. Actually here I copied the logic of converting autorest LRO: https://github.com/Azure/autorest.csharp/blob/bdc076cfe6dcfb1003416c9926c08039a5a05005/src/AutoRest.CSharp/Common/Input/CodeModelPartials.cs#L74-L92

Reply:
> what is final response? how do we use it? Must every LRO have final response?

@chunyu3 This should be the return value of a success LRO. It doesn't necessarily return a value. We support HTTP `204` which returns no content. In the test project, I defined two operations: one has return value, another doesn't.

Reply:
If we choose `200` when both `200` and `204` exist. But I am afraid that we cannot make sure that `200` response is always before `204` response in cadl (there is no rule to force the response order). so it is possible that the `204` response will be return as final response if `204` response is before `200` in `Response` array.

Reply:
I'll update the logic to find all supported return responses, and return the best match.